appcues/src/main/java/com/appcues/DeepLinkHandler.kt

```diff
@@ -66,4 +77,23 @@ internal class DeepLinkHandler(
             else -> false
         }
     }
+
+    private suspend fun previewExperience(experienceId: String, activity: Activity) {
+        experienceRenderer.preview(experienceId).run {
+            val resources = activity.resources
+
+            when (this) {
+                is HttpError -> if (code == NETWORK_ERROR_NOT_FOUND) {
+                    resources.getString(R.string.appcues_preview_flow_not_found)
+                } else {
+                    resources.getString(R.string.appcues_preview_flow_failed)
+                }
```

Comment:
Might be best suited to do the 404 check down in the ExperienceRenderer when the networking error is being handled, that way the logic is centralized and not the responsibility of any callers? There could be separate `RenderingError` options for the not found vs other case then, which bubble up to the caller.